baseline/tf/classify/training/datasets.py

```diff
@@ -145,3 +145,125 @@ def fit_datasets(model_params, ts, vs, es=None, **kwargs):
         trainer.recover_last_checkpoint()
         trainer.sess.run(test_init_op)
         trainer.test(es, reporting_fns, phase='Test', verbose=verbose)
+
+
+@register_training_func('classify', name="multi-label-datasets")
```

Comment:
why is this needed?  it doesnt seem to do anything different from the base

Reply:
in general, I think we should strive to avoid custom `fit_func`s and try and limit them to cases which involve a different sub-system (like eager, distributed, datasets, feed_dict etc.), even if it means some casing within the class.  For example, here you have only overriden the TF 1.x fit for datasets, so does this mean that only this mode supports multi-label?

Reply:
This function needed because of this line

```
model_params['y'] = y
```

In the default implementation it tries to turn the `y` into a one hot vector

```
model_params['y'] = tf.one_hot(tf.reshape(y, [-1]), len(model_params['labels']))
```

But when you are multlihot your `y` is already coming in a a `[B, C]` tensor.

I suppose it would be possible to detect if it was needed based on the shape of `y`? This seems like it would also allow for people to have non sparse labels in the single label situation, like if they were label smoothing or something?

Reply:
The section of the code where this change is made is analogous to the `make_input` function for the feed_dict code right? I wonder if this part should be customizable via some sort of hook?